src/lib.rs

```diff
@@ -1,2 +1,40 @@
 mod graphql;
+pub mod models;
 pub mod papr_subgraph;
+pub mod schema;
+
+use self::models::{NewTwab, Twab};
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use dotenvy::dotenv;
+use reservoir_nft::oracle::OracleResponse;
+use std::env;
+
+pub fn establish_connection() -> PgConnection {
+    dotenv().ok();
+
+    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
+    PgConnection::establish(&database_url)
+        .unwrap_or_else(|_| panic!("Error connecting to {}", database_url))
+}
+
+pub fn add_twabs(conn: &mut PgConnection, oracle_responses: &Vec<OracleResponse>) -> Vec<Twab> {
+    use crate::schema::twabs;
+
+    let mut n: Vec<NewTwab> = Vec::new();
+    for res in oracle_responses {
+        n.push(NewTwab {
+            token_address: &res.message.id,
+            // TODO: figure out how to get currency address
+            currency_address: &"",
```

Comment:
@adamgobes do you know how to decode the payload to get this? @wilsoncusack ?

Reply:
maybe not worth doing now on second thought

Reply:
I think we could hard code to 0x0 for now or have a conditional on what we sent to the oracle fn